p2-maven-plugin/src/main/java/org/eclipse/tycho/p2maven/ProjectDependencyClosureGraph.java

```diff
@@ -13,39 +13,55 @@
 package org.eclipse.tycho.p2maven;
 
 import java.io.File;
+import java.io.FileWriter;
 import java.io.IOException;
+import java.io.PrintWriter;
 import java.util.AbstractMap.SimpleEntry;
+import java.util.ArrayList;
 import java.util.Collection;
 import java.util.Collections;
 import java.util.HashMap;
+import java.util.HashSet;
+import java.util.LinkedHashMap;
+import java.util.LinkedHashSet;
 import java.util.List;
 import java.util.Map;
 import java.util.Map.Entry;
 import java.util.Optional;
 import java.util.Set;
-import java.util.concurrent.ConcurrentHashMap;
-import java.util.concurrent.CopyOnWriteArrayList;
 import java.util.function.Function;
 import java.util.stream.Collectors;
 import java.util.stream.Stream;
 
 import org.apache.maven.project.MavenProject;
 import org.eclipse.core.runtime.CoreException;
-import org.eclipse.core.runtime.IStatus;
-import org.eclipse.core.runtime.MultiStatus;
-import org.eclipse.equinox.internal.p2.metadata.InstallableUnit;
 import org.eclipse.equinox.p2.metadata.IInstallableUnit;
 import org.eclipse.equinox.p2.metadata.IProvidedCapability;
 import org.eclipse.equinox.p2.metadata.IRequirement;
-import org.eclipse.equinox.p2.query.CollectionResult;
-import org.eclipse.equinox.p2.query.IQueryable;
 import org.eclipse.tycho.p2maven.MavenProjectDependencyProcessor.ProjectDependencies;
 import org.eclipse.tycho.p2maven.MavenProjectDependencyProcessor.ProjectDependencyClosure;
-import org.eclipse.tycho.p2maven.io.MetadataIO;
 import org.eclipse.tycho.p2maven.tmp.BundlesAction;
 
 class ProjectDependencyClosureGraph implements ProjectDependencyClosure {
 
+	/**
+	 * Represents a requirement from an installable unit
+	 */
+	record Requirement(IInstallableUnit installableUnit, IRequirement requirement) {
+	}
+
+	/**
+	 * Represents a capability provided by an installable unit
+	 */
+	record Capability(IInstallableUnit installableUnit, IProvidedCapability providedCapability) {
```

Comment:
```suggestion
	record Capability(MavenProject, project, IInstallableUnit installableUnit) {
```

We should record what project provides this as we later want to check for example if an edge point back to our self and similar. This will then also simplify computeProjectDependenciesFromGraph

Also the capabilty must be removed here for now, we can't really say (with public API) that a particular capbilty satiisfies a requrement. A requirment can only be satisfied by a whole IU. THis is becuase a requirement can be an expression that checks for multiple cpabilties of an IU